python/ground_git.py

```diff
@@ -0,0 +1,1056 @@
+                    if (('sourceKey' in fileDict) and (fileDict['sourceKey'] == sourceKey)
+                        and (fileDict['class'] == className)):
+                        return fileDict
+
+    def _read_version(self, id, className):
+        files = [f for f in os.listdir(self.path) if os.path.isfile(os.path.join(self.path, f))]
+        for file in files:
+            filename = file.split('.')
+            if (filename[-1] == 'json') and (filename[0] == str(id)):
+                with open(self.path + file, 'r') as f:
+                    fileDict = json.loads(f.read())
+                    if (fileDict['class'] == className):
+                        return fileDict
+
+    def _read_all_version(self, sourceKey, className, baseClassName):
+        baseId = (self._read_files(sourceKey, baseClassName))['id']
+        baseIdName = baseClassName[:1].lower() + baseClassName[1:] + "Id"
+
+        versions = {}
+        files = [f for f in os.listdir(self.path) if os.path.isfile(os.path.join(self.path, f))]
+        for file in files:
+            filename = file.split('.')
+            if (filename[-1] == 'json') and (filename[0] != 'ids'):
+                with open(self.path + file, 'r') as f:
+                    fileDict = json.loads(f.read())
+                    if ((baseIdName in fileDict) and (fileDict[baseIdName] == baseId)
+                        and (fileDict['class'] == className)):
+                        versions[fileDict['id']] = fileDict
+        return versions
+
+    def _read_all_version_ever(self, className):
+        versions = {}
+        files = [f for f in os.listdir(self.path) if os.path.isfile(os.path.join(self.path, f))]
+        for file in files:
+            filename = file.split('.')
+            if (filename[-1] == 'json') and (filename[0] != 'ids'):
+                with open(self.path + file, 'r') as f:
+                    fileDict = json.loads(f.read())
+                    if (fileDict['class'] == className):
+                        versions[fileDict['id']] = fileDict
+        return versions
+
+    def _find_file(self, sourceKey, className):
+        files = [f for f in os.listdir(self.path) if os.path.isfile(os.path.join(self.path, f))]
+        for file in files:
+            filename = file.split('.')
+            if (filename[-1] == 'json') and (filename[0] != 'ids'):
+                with open(self.path + file, 'r') as f:
+                    fileDict = json.loads(f.read())
+                    if (('sourceKey' in fileDict) and (fileDict['sourceKey'] == sourceKey)
+                        and (fileDict['class'] == className)):
+                        return True
+        return False
+
+    def __run_proc__(self, bashCommand):
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return str(output, 'UTF-8')
+
+    def _check_init(self):
+        if(not self.initialized):
```

Comment:
What happens if I created a repo a long time ago and am reusing it now? How do I know that it's been initialized?

Reply:
the .init() method will automatically check for previous initialization and will call init again which is safe but will not replace the repo and your existing work. The ids will carry on with the highest id and continue, not replacing older files. However, this implementation does not handle the case where the folder has been previously manually created, then initialized to some other repo with previous commits that have no relation to the ground objects, it will just continue to commit to that initial repo.

Reply:
That's fine for now. Let's add a `README.md` to the directory with documentation like this.